collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -277,21 +273,28 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
                 questionWidget = new TriggerWidget(context, fep);
                 break;
             case Constants.CONTROL_RANGE:
-                switch (fep.getDataType()) {
-                    case Constants.DATATYPE_INTEGER:
-                        questionWidget = new RangeIntegerWidget(context, fep);
-                        break;
-                    case Constants.DATATYPE_DECIMAL:
-                        questionWidget = new RangeDecimalWidget(context, fep);
-                        break;
+
+                //questionWidget = null;
+
+                if (appearance.startsWith("rating")) {
+                    questionWidget = new RatingWidget(context, fep);
+                } else {
+                    switch (fep.getDataType()) {
+                        case Constants.DATATYPE_INTEGER:
+                            questionWidget = new RangeIntegerWidget(context, fep);
+                            break;
+                        case Constants.DATATYPE_DECIMAL:
+                            questionWidget = new RangeDecimalWidget(context, fep);
+                            break;
+                        default:
+                            questionWidget = new StringWidget(context, fep, readOnlyOverride);
+                            break;
+                    }
+                }
+                    break;
                     default:
                         questionWidget = new StringWidget(context, fep, readOnlyOverride);
                         break;
-                }
-                break;
-            default:
```

Comment:
This default case is important, though, because it's a catch-all for unrecognized types. Why did you remove it?

Reply:
Looks like I made some unintentional changes. Will correct it.